src/platform_impl/windows/event_loop.rs

```diff
@@ -521,103 +541,111 @@ lazy_static! {
+            RegisterWindowMessageA("Tao::WakeupMsg")
         }
     };
     // Message sent when we want to execute a closure in the thread.
     // WPARAM contains a Box<Box<dyn FnMut()>> that must be retrieved with `Box::from_raw`,
     // and LPARAM is unused.
     static ref EXEC_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::ExecMsg\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Tao::ExecMsg")
         }
     };
     static ref PROCESS_NEW_EVENTS_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::ProcessNewEvents\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Tao::ProcessNewEvents")
         }
     };
     /// lparam is the wait thread's message id.
     static ref SEND_WAIT_THREAD_ID_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::SendWaitThreadId\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Tao::SendWaitThreadId")
         }
     };
     /// lparam points to a `Box<Instant>` signifying the time `PROCESS_NEW_EVENTS_MSG_ID` should
     /// be sent.
     static ref WAIT_UNTIL_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::WaitUntil\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Tao::WaitUntil")
         }
     };
     static ref CANCEL_WAIT_UNTIL_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::CancelWaitUntil\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Tao::CancelWaitUntil")
         }
     };
     // Message sent by a `Window` when it wants to be destroyed by the main thread.
     // WPARAM and LPARAM are unused.
     pub static ref DESTROY_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Tao::DestroyMsg\0".as_ptr() as LPCSTR)
+            RegisterWindowMessageA("Tao::DestroyMsg")
         }
     };
     // WPARAM is a bool specifying the `WindowFlags::MARKER_RETAIN_STATE_ON_SIZE` flag. See the
     // documentation in the `window_state` module for more information.
     pub static ref SET_RETAIN_STATE_ON_SIZE_MSG_ID: u32 = unsafe {
-        winuser::RegisterWindowMessageA("Tao::SetRetainMaximized\0".as_ptr() as LPCSTR)
+        RegisterWindowMessageA("Tao::SetRetainMaximized")
     };
     static ref THREAD_EVENT_TARGET_WINDOW_CLASS: Vec<u16> = unsafe {
+        let mut class_name= util::to_wstring("Tao Thread Event Target");
 
-        let class_name= util::to_wstring("Tao Thread Event Target");
-
-        let class = winuser::WNDCLASSEXW {
-            cbSize: mem::size_of::<winuser::WNDCLASSEXW>() as UINT,
-            style: 0,
-            lpfnWndProc: Some(winuser::DefWindowProcW),
+        let class = WNDCLASSEXW {
+            cbSize: mem::size_of::<WNDCLASSEXW>() as u32,
+            style: WNDCLASS_STYLES(0),
+            lpfnWndProc: Some(call_default_window_proc),
```

Comment:
The bindings don't quite match the expected type. One is an `fn` type which takes the parameter types without conversion, and the binding for the specific exported function has `IntoParam` generic trait types for the parameters which turn into a pass through if the type already matches but can also convert more idiomatic Rust types like `&str` automatically when you call it.

Seems like there ought to be a conversion to raw `fn` for this scenario, it may even exist already. If not, I'll suggest it for `windows-rs `.

Reply:
Alright, can we move this `call_default_window_proc` function to util module and use it instead of defining it multiple times.